File: files/400-499/438/CH-438.py
import pandas as pd
import math

path = "400-499/438/CH-438 Number Puzzles - Spy Number.xlsx"
input = pd.read_excel(path, usecols="B", nrows=8, skiprows=2)
test = pd.read_excel(path, usecols="D", nrows=8, skiprows=2)

result = [
    n
    for n in range(10000, 20001)
    if sum(int(x) for x in str(n)) == math.prod(int(x) for x in str(n))
][:7]

File: files/400-499/438/CH-438.R
library(tidyverse)
library(readxl)

path <- "400-499/438/CH-438 Number Puzzles - Spy Number.xlsx"

is_spy <- function(num) {
  spl <- str_split(as.character(num), "")[[1]] %>% as.numeric()
  sum(spl) == prod(spl)
}

result <- tibble(n = 10000:20000) %>%
  mutate(is_spy = map_lgl(n, is_spy)) %>%
  filter(is_spy) %>%
  head(7)
